sharding-spring/sharding-jdbc-spring/sharding-jdbc-spring-boot-starter/src/main/java/org/apache/shardingsphere/shardingjdbc/spring/boot/SpringBootConfiguration.java

```diff
@@ -148,7 +156,13 @@ private DataSource getDataSource(final Environment environment, final String pre
         if (dataSourceProps.containsKey(jndiName)) {
             return getJndiDataSource(dataSourceProps.get(jndiName).toString());
         }
-        return DataSourceUtil.getDataSource(dataSourceProps.get("type").toString(), dataSourceProps);
+        String type = dataSourceProps.get("type").toString();
+        DataSource datasource = DataSourceUtil.getDataSource(type, dataSourceProps);
+        DataSourcePropertiesSetter dataSourcePropertiesSetter = DATA_SOURCE_PROPERTIES_SETTER_MAP.get(type);
+        if (null != dataSourcePropertiesSetter) {
+            dataSourcePropertiesSetter.propertiesSet(environment, prefix, dataSourceName, datasource);
+        }
+        return datasource;
```

Comment:
please rename `datasource` to `result` if the variable is return value of method